tools/dtrace/slowest-queries.d

```diff
@@ -0,0 +1,53 @@
+#!/usr/sbin/dtrace -qZs
+
+#pragma strsize=4k
+
+/* Print out the slowest query every 10 seconds. */
+
+struct query_info {
+    string conn_id;
+    string query;
+    uint64_t latency;
+};
+
+struct query_info slowest_query;
+self struct query_info queries[string];
+
+dtrace:::BEGIN
+{
+    printf("Tracing slowest queries for nexus PID %d, use Ctrl-C to exit\n", $target);
+}
+
+diesel_db$target:::query_start
+{
+    this->conn_id = json(copyinstr(arg1), "ok");
+    self->queries[this->conn_id].conn_id = json(copyinstr(arg1), "ok");
```

Comment:
```suggestion
    self->queries[this->conn_id].conn_id = this->conn_id```

Reply:
also we're never deallocating this so eventually we'll run DTrace out of dynamic variable memory

Reply:
That sounds bad :) I guess we'd need to clear this in the `tick-10s` action?

Reply:
typically you'll need to clear the (non-aggregation variable) state you create in the start probe in the done probe